tests/Unit/Domain/Test_Orientations.cpp

```diff
@@ -0,0 +1,181 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
```

Comment:
My argument is that if it makes sense to copy something, a value operation, (in most cases) it is equally reasonable to check if the values are equal. We may encounter exceptions, but I'd be surprised if there are many. Typically for such classes the equivalence operator will be an `and` between the data members anyway. I don't think we need a hard rule "do this for everything", but I think it is good to have it as a general suggestion, where not doing adding equivalence operators should be the exception and maybe even a C++ comment ask people to add a comment why it makes sense to copy a class but not check if they are equal.